engine/message/db.go

```diff
@@ -264,82 +265,89 @@ func NewDB(ctx context.Context, db *sql.DB, c *Config) (*DB, error) {
 				cm.id = msg.contact_method_id and
 				cm.disabled
 		`),
-		pending: p.P(fmt.Sprintf(`
+
+		messages: p.P(`
 			select
 				msg.id,
 				msg.message_type,
-				msg.contact_method_id,
 				cm.type,
+				chan.type,
+				coalesce(msg.contact_method_id, msg.channel_id),
+				coalesce(cm.value, chan.value),
 				msg.alert_id,
 				msg.alert_log_id,
 				msg.user_verification_code_id,
-				msg.channel_id,
-				chan.type
+				msg.user_id,
+				msg.service_id,
+				msg.created_at,
+				msg.sent_at
 			from outgoing_messages msg
 			left join user_contact_methods cm on cm.id = msg.contact_method_id
 			left join notification_channels chan on chan.id = msg.channel_id
-			where last_status = 'pending' and (not cm isnull or not chan isnull)
-			order by
-				msg.message_type,
-				(select max(sent_at) from outgoing_messages om where om.escalation_policy_id = msg.escalation_policy_id) nulls first,
-				(select max(sent_at) from outgoing_messages om where om.service_id = msg.service_id) nulls first,
-				(select max(sent_at) from outgoing_messages om where om.alert_id = msg.alert_id) nulls first,
-				channel_id,
-				(select max(sent_at) from outgoing_messages om where om.user_id = msg.user_id) nulls first,
-				(select max(sent_at) from outgoing_messages om where om.contact_method_id = msg.contact_method_id) nulls first,
-				msg.created_at,
-				msg.alert_id,
-				msg.alert_log_id,
-				msg.contact_method_id
-			limit %d
-		`, c.MaxMessagesPerCycle)),
+			where
+				sent_at > now() - '10 minutes'::interval or
+				last_status = 'pending' and
+				(msg.contact_method_id isnull or msg.message_type = 'verification_message' or not cm.disabled)
+		`),
 	}, p.Err
 }
 
-func (db *DB) getRows(ctx context.Context, tx *sql.Tx) ([]row, error) {
-	rows, err := tx.Stmt(db.pending).QueryContext(ctx)
+func (db *DB) currentQueue(ctx context.Context, tx *sql.Tx, now time.Time) (*queue, error) {
+	rows, err := tx.Stmt(db.messages).QueryContext(ctx)
 	if err != nil {
 		return nil, errors.Wrap(err, "fetch outgoing messages")
 	}
 	defer rows.Close()
 
-	var result []row
-	var r row
+	var result []Message
 	for rows.Next() {
-		var cmID, cmType, chID, chType sql.NullString
+		var msg Message
+		var cID, cValue, verifyID, userID, serviceID, cmType, chanType sql.NullString
```

Comment:
It refers to either channel ID _or_ contact method ID based on the type (which isn't known until after scanning). Same for `cValue` it can be the channel value or the contact method value depending on type.

Only `cmType` and `chanType` are distinct.

Reply:
Ah, is there some way we can denote that? Variable names in Go can be hard to follow, especially while reviewing/revisiting.

Reply:
Hmm... I think `destID` and `destValue` should work -- They are only used once a few lines down as `msg.Dest.ID` and `msg.Dest.Value`, respectively.